### DISCUSSION TITLE
Empty build directories after quitting `npm run dev`

### DISCUSSION THREAD

**Discussion by jensmeindertsma** [remix-run/remix#2922] [Proposals]
When you run `npx remix run` and then terminate the server with `Ctrl-C`, the following code is executed:
https://github.com/remix-run/remix/blob/c1821e4f2c7e844f3b7462bbd6ae7711479c1c7d/packages/remix-dev/cli/commands.ts#L91-L94
However, if you add a script to your `package.json` called `dev` which runs `remix run`, when you terminate the server after running `npm run dev`, this code is never called.

It would be nice if the build directories were emptied even when running `remix run` through a NPM script.

**Comment by sergiodxa** [comment]
I run Remix run inside an npm script inside PM2 (PM2 runs npm run dev) and it works correctly the delete of the build directory

**Comment by jensmeindertsma** [comment]
https://user-images.githubusercontent.com/64677517/133921602-81ac002d-50d2-482d-974a-ed329bd5211e.mov

Here you can see the behavior I'm referring to. Pay attention to the `build` folder on the right.

**Comment by jensmeindertsma** [comment]
Here is a patch file for fixing the issue. It seems that `signal-exit` isn't doing it's job. Might be worth ripping that out in favour of just `process.on`?

```diff
diff --git a/node_modules/@remix-run/dev/cli/commands.js b/node_modules/@remix-run/dev/cli/commands.js
index 2d6c2b2..460046b 100644
--- a/node_modules/@remix-run/dev/cli/commands.js
+++ b/node_modules/@remix-run/dev/cli/commands.js
@@ -89,9 +89,10 @@ async function watch(remixRootOrConfig, modeArg, onRebuildStart) {
     }
 
   }));
-  signalExit__default['default'](() => {
+  process.on("SIGINT", () => {
     fse.emptyDirSync(config$1.assetsBuildDirectory);
     fse.emptyDirSync(config$1.serverBuildDirectory);
+    process.exit(0);
   });
   console.log(`💿 Built in ${prettyMs__default['default'](Date.now() - start)}`);
 }

```

**Comment by jensmeindertsma** [comment]
Please look at my reproduction here: https://github.com/jensmeindertsma/using-signal-exit

I've demonstrated there that the issue has to do with `signal-exit`, and it not calling the callback function when the process is run as a child process of NPM.

**Comment by jensmeindertsma** [comment]
I've now added a demo of `process.on` to my reproduction
